internal/api/hooks.go

```diff
@@ -55,20 +74,176 @@ func (a *API) runPostgresHook(ctx context.Context, tx *storage.Connection, name
 	return response, nil
 }
 
-// invokeHook invokes the hook code. tx can be nil, in which case a new
+func readBodyWithLimit(rsp *http.Response) ([]byte, error) {
```

Comment:
There's a simpler algorithm for this based on HTTP semantics.

Just read the Response's Content-Length. If there's no such length, then it's not acceptable. Check the size of the Content-Length, and then if it's acceptable, put it in the LimitedReader. No need to do all of this.

You can also pipe the limited reader directly into the JSON decoder so no need to do byte reading etc.

Reply:
The developer can send over an incorrect `Content-Length` though. Wouldn't it also mean that the developer has to attach a `Content-Length` header?

Will explore the JSON Decoder